public/docs/_examples/dependency-injection/dart/lib/heroes/hero_service_provider.dart

```diff
@@ -0,0 +1,18 @@
+// #docregion
+
+import "package:angular2/core.dart";
+import "hero_service.dart";
+import "../logger_service.dart";
+import "../user_service.dart";
+
+// #docregion factory
+heroServiceFactory(Logger logger, UserService userService) =>
```

Comment:
I wonder whether [Dart factory constructors](https://www.dartlang.org/docs/dart-up-and-running/ch02.html#factory-constructors) might be a Dartier way to do this. @zoechi?